packages/desktop-client/src/components/App.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import {
+  init as initConnection,
+  send,
+} from 'loot-core/src/platform/client/fetch';
+import { type GlobalPrefs } from 'loot-core/src/types/prefs';
+
+import { useActions } from '../hooks/useActions';
+import installPolyfills from '../polyfills';
+import { ResponsiveProvider } from '../ResponsiveProvider';
+import { styles, hasHiddenScrollbars, ThemeStyle } from '../style';
+
+import AppBackground from './AppBackground';
+import View from './common/View';
+import DevelopmentTopBar from './DevelopmentTopBar';
+import ErrorBoundary from './ErrorBoundary';
+import FatalError from './FatalError';
+import FinancesApp from './FinancesApp';
+import ManagementApp from './manager/ManagementApp';
+import MobileWebMessage from './MobileWebMessage';
+import UpdateNotification from './UpdateNotification';
+
+type AppProps = {
+  budgetId: string;
+  cloudFileId: string;
+  loadingText: string;
+  loadBudget: (
+    id: string,
+    loadingText?: string,
+    options?: object,
+  ) => Promise<void>;
+  closeBudget: () => Promise<void>;
+  loadGlobalPrefs: () => Promise<GlobalPrefs>;
+};
+
+function App({
+  budgetId,
+  cloudFileId,
+  loadingText,
+  loadBudget,
+  closeBudget,
+  loadGlobalPrefs,
+}: AppProps) {
+  const [initializing, setInitializing] = useState(true);
+  const [hiddenScrollbars, setHiddenScrollbars] = useState(
+    hasHiddenScrollbars(),
+  );
+
+  async function init() {
+    const socketName = await global.Actual.getServerSocket();
+
+    try {
+      await initConnection(socketName);
+    } catch (e) {
+      if (e.type === 'app-init-failure') {
+        setInitializing(false);
+        return;
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I think we should throw an exception here.

Before: `fatalError` gets set and we show error message

Now: `fatalError` is not set - do we show an error? Don't think we do.

Reply:
Updated to show the error UI on init error